src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1625,97 +1649,235 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 });
 
                 return;
             }
 
-            if (ticket == null) {
-                Options.Logger.WriteError("invalid token");
+            //TODO: Re-examine authentication logic to ensure authentication happens in some way.
+            //TODO: Allow extensibility to skip authentication somehow for access tokens.
+            //      Perhaps in Options.Provider.ValidationEndpoint()?
+            // Attempt to Recieve an Access Token for authentication
+            // See https://tools.ietf.org/html/rfc7662#section-2.1
+            var authTicket = await ReceiveAccessTokenAsync(request.Token, request);
+
+            // If the access token could not be recieved or the ticket could not be parsed,
+            // Attempt to validate based on client credentials
+            if(authTicket == null){
+                var clientNotification = new ValidateClientAuthenticationContext(Context, Options, request);
+                await Options.Provider.ValidateClientAuthentication(clientNotification);
+    
+                // Reject the request if client authentication was rejected.
+                if (!clientNotification.IsValidated)
+                {
+                    Options.Logger.WriteError("invalid client authentication.");
+    
+                    await SendErrorPayloadAsync(new OpenIdConnectMessage
+                    {
+                        Error = clientNotification.Error ?? OpenIdConnectConstants.Errors.InvalidClient,
+                        ErrorDescription = clientNotification.ErrorDescription,
+                        ErrorUri = clientNotification.ErrorUri
+                    });
+    
+                    return;
+                }
+    
+                // Reject if client authentication was skipped.
+                if (clientNotification.IsSkipped && request.IsClientCredentialsGrantType())
+                {
+                    Options.Logger.WriteError("client authentication is required for client_credentials grant type.");
+    
+                    await SendErrorPayloadAsync(new OpenIdConnectMessage
+                    {
+                        Error = OpenIdConnectConstants.Errors.InvalidGrant,
+                        ErrorDescription = "client authentication is required when using client_credentials"
+                    });
+    
+                    return;
+                }
+            }
 
-                await SendErrorPayloadAsync(new OpenIdConnectMessage {
-                    Error = OpenIdConnectConstants.Errors.InvalidGrant,
-                    ErrorDescription = "Invalid access token received"
-                });
+            // Note: use the token_type_hint parameter to look up a token
+            // See https://tools.ietf.org/html/rfc7662#section-2.1
+            AuthenticationTicket ticket = null;
+            string hint;
+            if (request.Parameters.TryGetValue(OpenIdConnectConstants.Parameters.TokenTypeHint, out hint))
```

Comment:
Good question. I guess I forgot about it! :stuck_out_tongue_closed_eyes: I'll make the change.